frontend/src/components/RoutePlanner.tsx

```diff
@@ -11,26 +11,36 @@ import {
   IonTitle,
   IonToolbar
 } from "@ionic/react";
-import React, { useEffect, useState } from "react";
+import { formatISO, isSameMinute, parseISO } from "date-fns";
+import React, { useState } from "react";
 import { Journey, Location } from "../api";
+import { useCurrentTime } from "../hooks/useCurrentTime";
+import { useCustomDepartureTimeUrlParamOrCurrentTime } from "../hooks/useCustomDepartureTimeOrCurrentTime";
 import { useJourneyQuery } from "../hooks/useJourneyQuery";
 import { useLocationByIdOrNull } from "../hooks/useLocationByIdOrNull";
 import { useStateParams } from "../hooks/useStateParams";
 import { IJourney } from "../interfaces/IJourney.interface";
 import { IJourneyStep } from "../interfaces/IJourneyStep.interface";
-import { useLocationFinderApi } from "../services/apiClients/ApiClientsContext";
 import { CreateFavoriteTrip, useFavoriteTrips } from "../services/favorites/FavoritesContext";
 import { FavoriteTripsComponent } from "./FavoriteTripsComponent";
 import JourneyListComponent from "./JourneyListComponent";
 import { LocationSearchInput } from "./LocationSearch/LocationSearchInput";
 
+/** String to represent the usage of current time in departure url param. */
+export const DEPARTURE_TIME_NOW_PARAM = "now";
+
 const RoutePlanner: React.FC = () => {
   const [originId, setOriginId] = useStateParams<string | null>(null, "origin", String, String);
   const [destinationId, setDestinationId] = useStateParams<string | null>(null, "destination", String, String);
 
   const originLocation = useLocationByIdOrNull(originId);
   const destinationLocation = useLocationByIdOrNull(destinationId);
 
+  const currentTime = useCurrentTime();
+  const [customDepartureTime, setCustomDepartureTime] = useStateParams<string>(DEPARTURE_TIME_NOW_PARAM, "departure", String, String);
```

Comment:
@Paulomart Is it intentional that initialValue passed to `useStateParams` is not represented in url when site is loaded? I would have expected to see `departure=now` in url when site is loaded (but maybe I just had some wrong expectations). As for the functionality it makes no difference here, just curious.